artifactory/utils/buildutils.go

```diff
@@ -274,3 +275,15 @@ func ValidateBuildAndModuleParams(buildConfig *BuildConfiguration) error {
 	}
 	return nil
 }
+
+// ReadAllContent stores the entire content of the given contentReader in the value pointed by the given pointer.
+func ReadAllContent(cr *content.ContentReader, pointer interface{}) error {
+	file, err := os.Open(cr.GetFilePath())
+	if err != nil {
+		return errorutils.CheckError(err)
+	}
+	byteValue, _ := ioutil.ReadAll(file)
+	file.Close()
```

Comment:
Let's do this in defer and put this line right after the error check.
Let's take this opportunity and make this change in other places where files are closed throughout the code base.